### PR TITLE
Improve test suite flakiness & randomized asan

### PR DESCRIPTION
**Pull request by bwoebi**
### Description

Add more retries to some tests to combat I/O related flakiness.
Fixes a nette test failing with 0.1% probability.
Fixes timeouts on internal-api-stress-test.php. (And a crash discovered by it.)

Adds randomized tests to asan and fixes the broken ASAN tests on master.

### Readiness checklist
- [ ] (only for Members) Changelog has been added to the release document.
- [ ] Tests added for this feature/bug.

### Reviewer checklist
- [ ] Appropriate labels assigned.
- [ ] Milestone is set.
- [ ] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: ext/span.c [hunk 1]
```diff
@@ -560,6 +561,7 @@ void ddtrace_drop_span(ddtrace_span_data *span) {
 
     if (stack->root_span == span) {
         ddtrace_switch_span_stack(stack->parent_stack);
+        stack->root_span = NULL;
```
Comment (morrisonlevi) [inline ext/span.c:564]: I assume these changes in `ext/span.c` are related to the failures in the randomized tests you found?

**Review by morrisonlevi — APPROVED** [review]
Approved, with some questions.

File: .circleci/config.yml [hunk 2]
```diff
@@ -657,7 +659,7 @@ jobs:
       - run:
           command: |
             mkdir -p /tmp/artifacts
-            find ~/datadog/tests -type f -name 'phpunit_error.log' -o -name 'nginx_*.log' -o -name 'php_fpm_*.log' -o -name 'dd_php_error.log' -exec cp --parents '{}' /tmp/artifacts \;
+            find ~/datadog/tests -type f \( -name 'phpunit_error.log' -o -name 'nginx_*.log' -o -name 'apache_*.log' -o -name 'php_fpm_*.log' -o -name 'dd_php_error.log' -o -name 'core' \) -exec cp --parents '{}' /tmp/artifacts \;
```
[the remark this answers is not in the document]
Comment (bwoebi) [reply .circleci/config.yml:662]: On Circleci I only have seen core, but adding the `*` likely does no harm.

File: .circleci/config.yml [hunk 3]
```diff
@@ -2511,12 +2598,22 @@ workflows:
...
           requires: [ 'package extension' ]
           matrix:
             parameters:
-              batch: [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ]
+              batch: [ 1, 2, 3, 4, 5 ]
+      - randomized_tests_asan:
+          requires: [ 'package extension zts-debug-asan' ]
+          matrix:
+            parameters:
+              batch: [ 6, 7, 8, 9, 10 ]
```
[the remark this answers is not in the document]
Comment (bwoebi) [reply .circleci/config.yml:2611]: Right, it's just to not have two jobs named the same.

File: ext/span.c (hunk 1 above)
Comment (bwoebi) [reply ext/span.c:564]: Yes :-)

**Pull request merged**